### DISCUSSION TITLE
S2-WL-ST USB

### DISCUSSION THREAD

**Discussion by dxmone** [alienatedsec/solis-ha-modbus-cloud#28] [General]
Hi, i just got my hybrid inverter and it's using a USB connection instead of the old 4 pin. Is it possible to do this solution with that inverster or am i screwed? Model: S6-EH3P10K-H

**Comment by dxmone** [comment]
I found out that the inverter has a RS485 dedicated port. So i guess i could connect direct to that instead of the logger?

**Reply by alienatedsec** [reply]
@dxmone The RS485 dedicated port could be for a meter in certain countries (e.g., the UK), but there is no harm in trying to connect the Waveshare device. I guess the port is also different and you will need to get the right plug from the internet.

**Reply by dxmone** [reply]
According to the manual it was just a normal ethernet and it provided a diagram. Is there any way to check if the waveshare module can read something from the inverter? 

(sorry it's in swedish)
![rs](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/36075893/cbf7363a-75ec-48cb-b1b2-d2ac50a1c80d)

**Reply by alienatedsec** [reply]
@dxmone That's easy then - make sure the A and B are connected correctly and we can find the way to read data :)

**Reply by dxmone** [reply]
I do think the wiring is correct. Do you have any good way to try to read the data? The port might not even be enabled default. The documentation is not the best.

**Reply by alienatedsec** [reply]
https://qmodbus.sourceforge.net/ try this

Connect with your waveshare and try reading some registers

**Reply by dxmone** [reply]
First time doing this with the waveshare. But the waveshare is connected to the network not by serial. Qmodbus can only connect using serial connection from what i can see? So how can my computer ask waveshare that is connected to the inverter?

**Reply by alienatedsec** [reply]
Use ModbusTCP section and Waveshare IP and port

**Reply by alienatedsec** [reply]
@dxmone 
![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/3a105a49-0997-4454-8571-e69d955e73ea)

**Reply by alienatedsec** [reply]
The ModbusTCP support started in version 0.3.0 https://github.com/ed-chemnitz/qmodbus/releases/tag/v0.3.0

**Reply by dxmone** [reply]
Thanks, i will try that out soon. My inverter has some problem that the installer must fix before i can start messing around.

**Reply by alienatedsec** [reply]
@dxmone any success?

**Comment by dxmone** [comment]
I got it to work just fine now. I had some wiring issues but now it's
working. Thanks for the help.

Den mån 29 jan. 2024 kl 18:39 skrev alienatedsec ***@***.***>:

**Reply by alienatedsec** [reply]
@dxmone 
How did you wire the `S2-WL-ST` USB datalogger? Can you share more details?

This could be useful for @mdeakin99 as per #31

**Comment by dxmone** [comment]
I wired it as the manual says. And then just connected A to A on the
waveshare and B o B. Then it just worked. After using the settings provided
by alienatedsecs guide.



Den lör 17 feb. 2024 kl 00:22 skrev alienatedsec ***@***.***>:

**Reply by alienatedsec** [reply]
@dxmone are you saying that the USB pinout on your inverter can be matched with the 4-pin socket from this project?

**Comment by dxmone** [comment]
Sorry i forgot that this question was about the usb device. I found out
that my solis inverter had a dedicated rs485 port so I used that one.

lör 17 feb. 2024 kl. 11:02 skrev alienatedsec ***@***.***>:

**Reply by frizzle0815** [reply]
Can you simultaneously use the Solis Datalogger via the USB port and the Waveshare device via the RS485 port?
So can we use Cloud an Home Assistant at the same time (which is the main purpose of this repo)?

The Solis Support states that this should not be possible, but at the same time, they mention that it seems to work for some customers regardless.

**Reply by dxmone** [reply]
Yes, i am using both at the same time.

**Reply by frizzle0815** [reply]
That sounds promising!
Does this mean that I don’t need two separate Waveshare server+client, but instead, I can use just one Waveshare (POE version) and just connect A+B and Network?

**Reply by dxmone** [reply]
that's how i am using it. Just one waveshare and controlling it only by homeassistant. But soliscloud works just as it should. Perfect from both worlds.

**Reply by frizzle0815** [reply]
Thank you for the prompt response!
My inverter will only be installed later this month and then I will also give it a try.

**Reply by frizzle0815** [reply]
My S6-EH3P10K-H-EU has finally been installed, but so far I haven't been able to read any registers. I can establish a connection to the Waveshare, but every attempt to receive data fails.

Do I need to activate the RS485 interface on the inverter separately somewhere? There appears to be an indicator on the front for when the interface is being used. This indicator does not light up on my device. Additionally, I measured the voltage between A and B and it is 0.0V. So I guess that the interface is not active at all...![Screenshot_20240616-212226.png](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/158803904/62842102-ec51-464f-ac85-a14dc6b95d2b)

**Reply by alienatedsec** [reply]
@frizzle0815 Did you connect the pins as instructed?

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/c9a67011-46b7-469f-b60a-5ad67b403b91)

Those need to go to Waveshare A and B respectively.

**Reply by frizzle0815** [reply]
I used a normal RJ45 CAT7 Cable and plugged it into the inverter. I cut off  the other end and connected blue to B and blue/white to A.
I even tried switching A and B without success.

I didn't add 120 Ohm resistor as often suggested but I guess that shouldn't be necessary because the cable is around 30cm long for testing purposes.

**Reply by alienatedsec** [reply]
@frizzle0815 I suggest contacting Solis via their ticketing system and asking how to enable this port. Nothing in the manual about this.

![image](https://github.com/alienatedsec/solis-ha-modbus-cloud/assets/73167064/b15ae2cb-302d-4527-82fb-deeb3ebfdff1)

**Reply by frizzle0815** [reply]
Thank you. I will report back if I found out anything.

**Reply by Sabnock2210** [reply]
Hallo. Ich habe nach tagelangem suchen hier vielleicht endlich meine Lösung gefunden. Leider bin ich nicht sehr fit in der ganzen Materie. Ich habe ebenfalls den Solis S6 3PH mit USB Datenlogger S2-WL-ST und einer separaten RS 485. Könnt ihr mir eine Liste schreiben was ich brauche um die Cloud und HA zu nutzen und wie ich das ganze verkabele? Bin absoluter beginner im Modbus

**Reply by frizzle0815** [reply]
Hallo Sabnock,
ich bin über den RS485 Ausgang des Solis S6 nicht weiter gekommen, habe aber für mich eine einfachere Lösung gefunden.
Du benötigst gar keinen separaten RS485 Konverter, sondern kannst direkt über den Datenlogger S2-WL-ST auslesen. Du musst also auch nichts zusätzlich verkabeln.
Einfach die IP-Adresse des Datenlogger verwenden und über Port 502 die benötigten Register abfragen. Das funktioniert bei mir problemlos, mit dem einzigen Nachteil, dass dann die Übertragung der Daten in die Solis Cloud regelmäßig unterbrochen wird. Das lässt sich vielleicht umgehen, wenn man kein allzu kurzes Abfrageintervall verwendet. Ich rufe die aktuellen Werte alle 10 Sekunden ab und dann klappt es mit der Solis Cloud zusätzlich nicht. Ich nutze die Solis App also nicht mehr, sondern mache meine eigenen Auswertungen.

**Reply by Sabnock2210** [reply]
Das funktioniert bei mir auch. Nur möchte ich die Cloud weiter nutzen um z.B. die Updates des WR zu bekommen.

**Reply by frizzle0815** [reply]
Die Cloud funktioniert damit meines Erachtens schon weiterhin - Updates solltest Du also bekommen.
Die Graphen in der Solis App haben nur teilweise lange Aussetzer, weil keine Messwerte abgerufen werden konnten. Es kann immer nur ein Modbus Teilnehmer gleichzeitig abfragen, weshalb es unter Umständen hilft, wenn Du über HA seltener die Werte aktualisierst. Ist für mich keine Lösung, weil ich stets aktuelle Werte für die Steuerung meiner Wallbox brauche.
Ich hatte da diverse Diskussionen mit dem Solis Support und bin zu keiner anderen Lösung gekommen - tut mir leid.

**Reply by Sabnock2210** [reply]
Komme mit dem Support auch nicht weiter. Ich habe eine Anleitung bekommen wie ich den Logger von RTU auf TCP umstelle. Da komme ich aber nicht weiter. 
Leider hängt sich bei mir die Cloud komplett auf, sobald ich den Logger für HA nutze. Der WR ist dann dauerhaft offline. Deshalb würde ich die beschriebene Lösung versuchen, nur bin ich der totale Anfänger in Sachen Modbus. Da brauche ich eine sehr detallierte Anleitung.

**Reply by alienatedsec** [reply]
> Da brauche ich eine sehr detallierte Anleitung.

Aren't the instructions on the main page detailed enough?

Sind die Anweisungen auf der Hauptseite nicht detailliert genug?

https://github.com/alienatedsec/solis-ha-modbus-cloud

**Reply by Sabnock2210** [reply]
The instructions are very good. I just don't understand them properly. According to the sketch, I need two Waveshares? a client and a server? then do I need an adapter for usb for the datalogger on the client? how do I tell the Waveshare that it is a client or server? Or do they work correctly if I set them up straight away?

**Reply by alienatedsec** [reply]
@Sabnock2210 Building based on this project requires some technical skills, and if those are not present, some self-learning by reading manuals and following some high-level instructions. Therefore, my recommendation is to start slowly and follow the diagram, search Google and eBay for terms listed in https://github.com/alienatedsec/solis-ha-modbus-cloud?tab=readme-ov-file#devices-and-cost

Besides that, consider a bigger box to fit all devices without hassle, as mine seems slightly too small.

**Reply by Sabnock2210** [reply]
I'll order the components and see if it works. I'll get in touch again then. Thank you for the support

**Reply by Sabnock2210** [reply]
The components are there. Unfortunately I can't get the firmware on it, I'm doing something wrong. VirCom says Flash write fail

**Reply by Sabnock2210** [reply]
Hello,
For now I've just set up the server. It's running!!!! Now I'm not looking for a solution to connect the S2 stick to the Waveshare with USB. I got a USB adapter to cable. Now connect 5v and then it should work

**Comment by DG-Lt** [comment]
Hi everyone,

Adding my experience to the conversation. Might be useful to others.
My inverter: Solis S6-EH3P20K-H
Solis Dongle: USB S2-WL-ST

I have been using [Solis Modbus](https://solis-modbus.readthedocs.io/en/latest/) integration on HA targeted to the Dongle IP. This way I got all the Modbus data to HA. Also the Solis Cloud App was sometimes refreshing the data as well (intermittently).

Yesterday received my Waveshare RS485 Gateway. and decided to switch to that [dedicated RS485](https://github.com/alienatedsec/solis-ha-modbus-cloud/discussions/28?sort=old#discussioncomment-8168546) as described here.
Spent some hours trying to establish the connection through the Blue and Blue/white wires. it did not work. was already thinking that the port was disabled, and not seeing any way to enable I thought I will need to contact Solis tech support to do that.

This is where I opened my inverter manual...
![image](https://github.com/user-attachments/assets/3981e35c-1a17-4d40-8e76-3a2043bb0276)

Switched to the Orange pair and it started working...

Connected
Solis Orange/White -> A on Waveshare gateway
Solis Orange            -> B on Waveshare gateway

So the message is: be aware that different Inverters might have different pinouts... Even of the similar model series...

**Reply by DG-Lt** [reply]
Another update.
Yesterday I finally spent time on removing the integration through USB S2-WL-S and switching the Waveshare on dedicated RS485 port on my solis S6-EH3P20K-H
 All went smoothly: first I removed the device with the datalogger IP. then added the device with the Waveshare IP. and renamed the device to the same that I had before

at first glance all seemed good.: same entity names and preserved statistics.

but today I noticed that out of many sensors some of them started reporting data differently:
![image](https://github.com/user-attachments/assets/18ce89fe-d11d-4f0e-aff9-f9d3fbf8851f)
![image](https://github.com/user-attachments/assets/a75cd04d-6908-49cf-8e22-68231f650d8b)

Battery Energy sensors seem to be off by 100x. or 10x
Seems that there is a bug somewhere. either inverter or [HA integration](https://github.com/Pho3niX90/solis_modbus).

This is messing up with my energy dashboard :(

**Reply by alienatedsec** [reply]
@DG-Lt The port you started using will likely report data with different metrics and integrations will need amending for such a method.

The best would be to test it with tools like QModBus, query all trouble registers and amend the integration parameters thereafter.

**Reply by DG-Lt** [reply]
It turned out that the problem was caused by the fact that I have chosen Waveshare as a connection type when added the device:
![image](https://github.com/user-attachments/assets/05fdfcf0-7fc9-4ada-a3cb-d8618e97de08)

It was explained [here ](https://github.com/Pho3niX90/solis_modbus/discussions/175#discussioncomment-12668480) that this should only be selected in some rare cases...

**Comment by fastbike** [comment]
> Hi, i just got my hybrid inverter and it's using a USB connection instead of the old 4 pin. Is it possible to do this solution with that inverster or am i screwed? Model: S6-EH3P10K-H

Hi I've just followed through this discussion. I've got the same inverter (S6-EH3P10K-H) but it has an AU suffix which is for the verification for the Australia / New Zealand market.
I'm trying to get this to work - initially just trying to read the Modbus registers using the Modbus Poll tool.
I can connect using a Waveshare RS485/ETH(B) bridge.
I keep getting timeouts. I've tried the blue/white wire pairs and the orange/white wire pairs and there is no difference.
I've followed the instructions in the main part of this repo but am out of ideas.

Am I screwed ?

**Comment by Sabnock2210** [comment]
Hello,

 

Unfortunately, I can't get that to work either. I only have one Waveshare connected. It worked for a few weeks. Now I haven't been able to connect for a long time and only use the portal.

 

I haven't found a solution yet to reconnect the USB stick.

 

 

Von: David Moorhouse ***@***.***> 
Gesendet: Mittwoch, 8. Oktober 2025 07:11
An: alienatedsec/solis-ha-modbus-cloud ***@***.***>
Cc: Sabnock2210 ***@***.***>; Mention ***@***.***>
Betreff: Re: [alienatedsec/solis-ha-modbus-cloud] S2-WL-ST USB (Discussion #28)

 

Hi, i just got my hybrid inverter and it's using a USB connection instead of the old 4 pin. Is it possible to do this solution with that inverster or am i screwed? Model: S6-EH3P10K-H

Hi I've just followed through this discussion. I've got the same inverter (S6-EH3P10K-H) but it has an AU suffix which is for the verification for the Australia / New Zealand market.
I'm trying to get this to work - initially just trying to read the Modbus registers using the Modbus Poll tool.
I can connect using a Waveshare RS485/ETH(B) bridge.
I keep getting timeouts. I've tried the blue/white wire pairs and the orange/white wire pairs and there is no difference.
I've followed the instructions in the main part of this repo but am out of ideas.

Am I screwed ?

**Reply by fastbike** [reply]
Which seems very strange because other people are reporting it working.

I have tried disconnecting the USB datalogger and plugging the RJ45 cable into the RS485 port. This is connected to the Waveshare device. I can see the activity light on the waveshare flashing as the ModbusPoll software tries to read the registers but the software keeps showing a time out.
Also I think the USB stick is actually using a USB protocol so despite users on other forums saying it should work as a client of another waveshare modbus bridge I can't see that working.
I have raised a service desk ticket with Solis and will report back if I get an answer.

**Comment by Sabnock2210** [comment]
That would be great. I would also like to make progress in this area.

 

 

Von: David Moorhouse ***@***.***> 
Gesendet: Freitag, 10. Oktober 2025 01:51
An: alienatedsec/solis-ha-modbus-cloud ***@***.***>
Cc: Sabnock2210 ***@***.***>; Mention ***@***.***>
Betreff: Re: [alienatedsec/solis-ha-modbus-cloud] S2-WL-ST USB (Discussion #28)

 

Which seems very strange because other people are reporting it working.

I have tried disconnecting the USB datalogger and plugging the RJ45 cable into the RS485 port. This is connected to the Waveshare device. I can see the activity light on the waveshare flashing at the ModbusPoll software tries to read the registers but the software keeps showing a time out.
Also I think the USB stick is actually using a USB protocol so despite users on other forums saying it should work as a client of another waveshare I can't see that working.
I have raised a service desk ticket with Solis and will report back if I get an answer.

**Reply by fastbike** [reply]
> That would be great. I would also like to make progress in this area. 

I've made some progress. I created an installer account with Solis, which gave me an organisation ID.
I then used the SolisCloud web page (Device | Modify Info) to set the Organisation Code.
After logging out from web page I was able to login using the installer account and then go the the Inverter Control page.
I now have access to the Parallel settings function - changing the Physical address ID to 2 seems to have enabled the RS485 port.
I'm now getting a different set of errors on the Modbus Poll tool, but at least it seems to be communicating.

**Comment by Sabnock2210** [comment]
I think that's a coincidence. Parallel settings are designed to connect multiple identical inverters.

 

 

Von: David Moorhouse ***@***.***> 
Gesendet: Samstag, 11. Oktober 2025 02:16
An: alienatedsec/solis-ha-modbus-cloud ***@***.***>
Cc: Sabnock2210 ***@***.***>; Mention ***@***.***>
Betreff: Re: [alienatedsec/solis-ha-modbus-cloud] S2-WL-ST USB (Discussion #28)

 

That would be great. I would also like to make progress in this area.

I've made some progress. I created an installer account with Solis, which gave me an organisation ID.
I then used the SolisCloud web page (Device | Modify Info) to set the Organisation Code.
After logging out from web page I was able to login using the installer account and then go the the Inverter Control page.
I now have access to the Parallel settings function - changing the Physical address ID to 2 seems to have enabled the RS485 port.
I'm now getting a different set of errors on the Modbus Poll tool, but at least it seems to be communicating.

**Reply by fastbike** [reply]
> I think that's a coincidence. Parallel settings are designed to connect multiple identical inverters. 

Indeed it was. The problem was the software I was using to test, QModBus could read a known register correctly.
It's early days but has been running overnight ok.
